### PR TITLE
Add support for more ac/fan modes

### PR DESCRIPTION
**Pull request by simontegelid**
Add support for settings fan modes and swing modes and another set of HVAC modes.

Based on my Qlima WMS S + SC52 (AB;AF)

### CODE REVIEW

**Comment by apbarratt** [comment]
Hopefully could be merged with this one too: https://github.com/rospogrigio/localtuya/pull/1177
Sadly there does not seem to be a priority on adding these new modes just yet, indeed, nothing has been merged since February :(

**Comment by jreenen** [comment]
@rospogrigio stated on #1213 the following:

> I don't like how this thing is being handled. We just can't keep adding presets as long as new commands pop out, each of them with their casing.
I believe we should turn this into a text edit field in which anyone would input the commands that are specific for their model, separated by a "/" or other char. The same applies also to many other plaforms (such as vacuums, etc...), we need to keep localtuya general purpose as much as we can, and refrain from introducing device-specific code or commands unless really impossible.

As I also submitted a pull request #1352 with some variation I do see his point. Currently, I am limited in time but if we want something merged we probably have more success by implementing the suggested refactoring instead of submitting variations, please correct me if I am wrong @rospogrigio

**Comment by simontegelid** [comment]
@jreenen I had missed that comment. I agree though, looks like a refactoring-PR is the way forward for this, #1177, #1213 and #1352

**Comment by jreenen** [comment]
@simontegelid are you ok if I use your PR to try to summarize some things?

@rospogrigio what would be your take on the following changes on behalf of the climate.py:
- Create a V3 of for the config flow
- For each enum value specified in [the built-in enum HVACMode](https://developers.home-assistant.io/docs/core/entity/climate/#hvac-modes) create a string config item to map the tuya action value.
- Refactor code to start using this mapping.
- Use [Config entry migration](https://developers.home-assistant.io/docs/config_entries_config_flow_handler#config-entry-migration) to migrate values to V3.

Almost sure I am missing some details here but could this be an approach you could agree on for your implementation?

**Comment by simontegelid** [comment]
Yes, please do!

**Comment by jmcollin78** [comment]
Waiting for this. Please merge is ok !

**Comment by pabsi** [comment]
Hello there!
I came across this PR as I recently acquired five BHT-006 units which all have heat/cold/fan(wind) modes. Sadly, I am unable to add them to Home Assistant because the climate changes here (or in other PRs are not merged yet). Now, with that said, I understand this is a personal project after all and @rospogrigio might not have time (or not enough anyway) to maintain it to a level probably thousands of us expect (105 PRs open, and last release from June 2023). But it'd be sad having to fork it (as I am seeing people doing) just because PRs are getting stuck. Perhaps, we can increase the contributors list? of course, if @rospogrigio agrees..

Other integrations, such as (Tuya Local)[https://github.com/make-all/tuya-local] have, for example, BHT-002 but not BHT-006 so I am in the same position there anyway (although I can add them in "heat only" mode, but not "cool" mode).

All this said, let me repeat I am very thankful for the integration and what allowed me to do so far at home, but I, as many others, just want to unblock myself from not being able to add some more devices to HA.

Thank you for all the hard work :pray:  and I hope my comment is not taken negatively.

**Review by rospogrigio — APPROVED** [review]

**Comment by simontegelid** [comment]
Thanks for your response! I'll fix conflicts asap

File: custom_components/localtuya/climate.py [hunk 1]
```diff
@@ -340,22 +432,14 @@ def min_temp(self):
         """Return the minimum temperature."""
         if self.has_config(CONF_MIN_TEMP_DP):
             return self.dps_conf(CONF_MIN_TEMP_DP)
-        # DEFAULT_MIN_TEMP is in C
-        if self.temperature_unit == UnitOfTemperature.FAHRENHEIT:
-            return DEFAULT_MIN_TEMP * 1.8 + 32
-        else:
-            return DEFAULT_MIN_TEMP
+        return self._config[CONF_TEMP_MIN]
```
Comment (simontegelid) [inline custom_components/localtuya/climate.py:435]: Not 100% sure about removing this conversion. I'm thinking that the CONF_TEMP_MIN is set in Fahrenheit during configuration if needed and conversion here is not needed. The same goes for CONF_TEMP_MAX of course.

**Comment by simontegelid** [comment]
@rospogrigio what are the next steps to move this forward?

**Comment by simontegelid** [comment]
Thanks @rospogrigio . What is the last CI step? Who are authorized to merge?

**Pull request merged**